protocol_rfcs/catalog-owned.md

```diff
@@ -0,0 +1,489 @@
+
+The catalog is responsible to implement an API (defined by the Delta client) that Delta clients can
+use to retrieve the latest ratified commit version (authoritative), as well as the set of ratified
+commits the catalog is still tracking for the table. If some commits needed to complete the snapshot
+are not tracked by the catalog, as they are already published, Delta clients can issue a filesystem
+LIST operation to retrieve them.
+
+Delta clients must establish an API contract where the catalog provides ratified commit information
+as part of the standard table resolution process performed at query planning time.
+
+## Publishing Commits
+
+Publishing is the process of copying the ratified commit with version `<v>` to
+`_delta_log/<v>.json`. The ratified commit may be a staged commit located in
+`_delta_log/_staged_commits/<v>.<uuid>.json`, or it may be an inline commit whose content the
+catalog tracks itself. Because the content of a ratified commit is immutable, it does not matter
+whether the client-side, server-side, or both catalog components initiate publishing.
+
+Implementations are strongly encouraged to publish commits promptly. This reduces the number of
+commits the catalog needs to track internally (and serve up to readers).
+
+Importantly, no ordering constraints apply to publishing. That is, version `v - 1` does not need to
+be published before version `v`. This will allow for parallel publishing and achieving higher
+throughput. It will enable catalog implementations to publish ratified commits right away without
+the need for synchronization or coordination between conurrent writers.
+
+Because of this, "gaps" may appears in the `_delta_log` in the filesystem: cases where versions `v`
+and `v + 2` are published, but version `v + 1` is not.
+
+**NOTE**: Because commit publishing can happen at any time and in any order after the commit
+succeeds, the file modification timestamp of the published file will not accurately reflect the
+original commit time. For this reason, catalog-owned tables must use
+[in-commit-timestamps](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#in-commit-timestamps)
+to ensure stability of time travel reads. Refer to
+[Writer Requirements for Catalog-Owned Tables](#writer-requirements-for-catalog-owned-tables)
+section for more details.
+
+## Maintenance Operations on Catalog-owned Tables
+
+Delta clients must follow the owning catalog's rules, if any, when performing maintenance
+operations like [Checkpoints](#checkpoints-1), [Log Compaction Files](#log-compaction-files),
+[Metadata Cleanup](#metadata-cleanup), and VACUUM. These rules can be unique to each catalog and may
+evolve over time.
+
+In particular, maintenance operations like Checkpoints, Log Compaction Files, and Metadata Cleanup
+can only be performed on published versions, unless the catalog specifies otherwise.
+
+To highlight: Because the catalog controls the lifetime and potentially the storage of the table,
```

Comment:
This is also not a catalog rule, it's a universal rule. If the catalog is OK with the clients vacuuming, then that's OK. Clients should not try to VACUUM the table without having the catalog's permission (according to whatever the catalog's rules are).